design/ironic-standalone-networking-service.md

```diff
@@ -0,0 +1,734 @@
+    DeviceType string `json:"deviceType,omitempty"`
+
+    // Credentials defines how to authenticate with the switch.
+    // +kubebuilder:validation:Required
+    Credentials SwitchCredentials `json:"credentials"`
+
+    // Port specifies the management port to connect to (e.g., SSH port 22, HTTPS port 443).
+    // If not specified, the driver will use its default port.
+    // +kubebuilder:validation:Minimum=1
+    // +kubebuilder:validation:Maximum=65535
+    // +optional
+    Port *int32 `json:"port,omitempty"`
+
+    // SkipTLSVerify disables verification of server certificates when using
+    // HTTPS to connect to the switch. This is required when the server
+    // certificate is self-signed, but is insecure because it allows a
+    // man-in-the-middle to intercept the connection.
+    // +optional
+    SkipTLSVerify *bool `json:"skipTLSVerify,omitempty"`
+}
+
+type BareMetalSwitchStatus struct {
+    // Conditions track the state of the switch configuration
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // LastUpdated is the timestamp when the switch configuration was last applied
+    LastUpdated *metav1.Time `json:"lastUpdated,omitempty"`
+}
+```
+
+#### BareMetalNetworkAttachment
+
+The `BareMetalNetworkAttachment` CRD defines how a switch port should be
+configured:
+
+```go
+// SwitchportMode defines the switchport mode for network interfaces.
+// +kubebuilder:validation:Enum=access;trunk;hybrid
+type SwitchportMode string
+
+const (
+    // SwitchportModeAccess sets the interface to access mode (single VLAN)
+    SwitchportModeAccess SwitchportMode = "access"
+    // SwitchportModeTrunk sets the interface to trunk mode (multiple VLANs)
+    SwitchportModeTrunk SwitchportMode = "trunk"
+    // SwitchportModeHybrid sets the interface to hybrid mode (access + trunk)
+    SwitchportModeHybrid SwitchportMode = "hybrid"
+)
+
+// BareMetalNetworkAttachmentSpec defines the desired switchport configuration.
+type BareMetalNetworkAttachmentSpec struct {
+    // Mode defines the switchport mode (access, trunk, or hybrid)
+    // +kubebuilder:validation:Enum=access;trunk;hybrid
+    // +kubebuilder:default=access
+    Mode SwitchportMode `json:"mode,omitempty"`
+
+    // NativeVLAN is the untagged VLAN ID for the port
+    // +kubebuilder:validation:Minimum=1
+    // +kubebuilder:validation:Maximum=4094
+    NativeVLAN int32 `json:"nativeVlan,omitempty"`
+
+    // AllowedVLANs is a list of VLAN IDs allowed on trunk/hybrid ports
+    // Only valid for trunk and hybrid modes
+    // +optional
+    AllowedVLANs []int32 `json:"allowedVlans,omitempty"`
+}
+
+type BareMetalNetworkAttachmentStatus struct {
+    // Conditions track the state of the network attachment
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // AttachedHosts lists the BareMetalHosts currently using this attachment
+    AttachedHosts []string `json:"attachedHosts,omitempty"`
```

Comment:
Do we have strong reasons not to use `OwnerReferences`? You want to avoid garbage collecting?